lib/testing-library-detection.ts

```diff
@@ -0,0 +1,52 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+/**
+ * Enhances a given rule with helpers to detect Testing Library utils.
+ */
+function testingLibraryDetection(rule: any) {
+  return (context: any, options: any) => {
+    let isImportingTestingLibrary = false;
+
+    // TODO: init here options based on shared ESLint config
+
+    // helpers for Testing Library detection
+    const helpers = {
+      getIsImportingTestingLibrary() {
+        return isImportingTestingLibrary;
+      },
+    };
+
+    // instructions for Testing Library detection
+    const detectionInstructions = {
+      ImportDeclaration(node: TSESTree.ImportDeclaration) {
+        isImportingTestingLibrary = /testing-library/g.test(
+          node.source.value as string
+        );
+      },
+    };
+
+    // update given rule to inject Testing Library detection
+    const ruleInstructions = rule(context, options, helpers);
+    const enhancedRuleInstructions = Object.assign({}, ruleInstructions);
+
+    Object.keys(detectionInstructions).forEach((instruction) => {
+      enhancedRuleInstructions[instruction] = (node: any) => {
+        if (instruction in detectionInstructions) {
+          detectionInstructions[instruction](node);
+        }
+
+        if (ruleInstructions[instruction]) {
+          return ruleInstructions[instruction](node);
+        }
+      };
+    });
+
+    return enhancedRuleInstructions;
+  };
+}
+
+function detect(rule: any) {
+  return testingLibraryDetection(rule);
+}
+
+export default detect;
```

Comment:
I'm a bit lost, but given default exports just take the name you want, we could just export `testingLibraryDetection`

if you wanted to enforce the name `detect`, then `detect` should be a named export. Unless to expect to add more code here 🤔

Reply:
This is a stupid function not necessary anymore. I first implemented this by calling `testingLibraryDetection` from this `detect` one using `bind` to pass the rule. Then I realized I could do the same just with a high-order function, but I forgot about removing this one! I'll get rid of it.